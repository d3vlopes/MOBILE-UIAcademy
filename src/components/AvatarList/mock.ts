import { AvatarListProps } from '.'

export const mockAvatar: AvatarListProps = {
  imageList: [
    {
      id: '1',
      imgUrl:
        'https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_04_nrxubl.jpg',
    },
    {
      id: '2',
      imgUrl:
        'https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_03_e3vugv.jpg',
    },
    {
      id: '3',
      imgUrl:
        'https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_05_ww4pum.jpg',
    },
    {
      id: '4',
      imgUrl:
        'https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_05_ww4pum.jpg',
    },
    {
      id: '5',
      imgUrl:
        'https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_05_ww4pum.jpg',
    },
  ],
}
