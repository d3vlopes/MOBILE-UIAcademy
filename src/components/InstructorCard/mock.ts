import { InstructorCardProps } from '.'

export const mockInstructors: InstructorCardProps[] = [
  {
    name: 'Pedro',
    avatar:
      'https://res.cloudinary.com/dbnq26wqe/image/upload/v1646189550/others/pedro_xoxvpu.png',
    work: 'UX â€¢ Carreira',
  },
  {
    name: 'Julia',
    avatar:
      'https://res.cloudinary.com/dbnq26wqe/image/upload/v1646189550/others/julia_xgq7kb.png',
    work: 'Mobile',
  },
  {
    name: 'Carlos',
    avatar:
      'https://res.cloudinary.com/dbnq26wqe/image/upload/v1646189550/others/carlos_sfgzrs.png',
    work: 'UI - Tipografia',
  },
]
