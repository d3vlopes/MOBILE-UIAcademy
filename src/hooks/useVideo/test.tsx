import React from 'react'
import { renderHook } from '@testing-library/react-hooks'
import { act } from 'react-test-renderer'

import { useVideo, VideoProvider, VideoProviderProps } from '.'

describe('useVideo', () => {
  it('should change the value of isFullScreen', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.changeFullScreenValue()
    })

    expect(result.current.isFullScreen).toBe(true)
  })

  it('should exec onPrevious function', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.onPrevious()
    })

    // expect()
  })

  it('should exec onNext function', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.onNext()
    })

    // expect()
  })
})
