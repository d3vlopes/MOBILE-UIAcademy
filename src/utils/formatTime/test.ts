import { formatTime } from '.'

describe('formatTime', () => {
  it('should return the minutes formatted', () => {
    const time = 45

    expect(formatTime(time)).toEqual('45min')
  })

  it('should return the hours formatted', () => {
    const time = 80

    expect(formatTime(time)).toEqual('1hr 20min')
  })
})
