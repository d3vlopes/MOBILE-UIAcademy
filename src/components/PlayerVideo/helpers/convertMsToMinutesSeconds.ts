import { padTo2Digits } from '../../../utils/formatTime/padTo2Digits'

export function convertMsToMinutesSeconds(milliseconds: number) {
  let minutes = Math.floor(milliseconds / 60000)
  let seconds = Math.floor((milliseconds % 60000) / 1000)

  if (Number.isNaN(minutes) || Number.isNaN(seconds)) {
    minutes = 0
    seconds = 0
  }

  return seconds === 60
    ? `${minutes + 1}:00`
    : `${minutes}:${padTo2Digits(seconds, 2)}`
}
