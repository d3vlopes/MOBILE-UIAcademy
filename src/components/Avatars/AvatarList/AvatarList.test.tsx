import React from 'react'

import { renderWithTheme } from '../../../utils/test/helpers'

import { mockAvatar } from './mock'

import { AvatarList } from '.'

describe('<AvatarList />', () => {
  it('should render 3 circles', () => {
    const { getAllByTestId } = renderWithTheme(
      <AvatarList imageList={mockAvatar.imageList.slice(0, 3)} />,
    )

    const wrapperAvatar = getAllByTestId('wrapperAvatar')

    expect(wrapperAvatar).toHaveLength(3)
  })

  it('should render circle with number of people', () => {
    const { getAllByTestId, getByTestId } = renderWithTheme(
      <AvatarList imageList={mockAvatar.imageList} />,
    )

    const wrapperAvatar = getAllByTestId('wrapperAvatar')
    const fillCircle = getByTestId('fillCircle')

    expect(wrapperAvatar).toHaveLength(3)
    expect(fillCircle).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { toJSON } = renderWithTheme(
      <AvatarList imageList={mockAvatar.imageList} />,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
