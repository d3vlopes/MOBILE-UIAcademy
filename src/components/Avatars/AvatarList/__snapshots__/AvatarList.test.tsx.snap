// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarList /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 25,
          "height": 108,
          "justifyContent": "center",
          "marginLeft": -20,
          "width": 108,
        },
      ]
    }
    testID="wrapperAvatar"
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_04_nrxubl.jpg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 98,
              "width": 98,
            },
          ]
        }
        testID="roundImage"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 25,
          "height": 108,
          "justifyContent": "center",
          "marginLeft": -20,
          "width": 108,
        },
      ]
    }
    testID="wrapperAvatar"
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_03_e3vugv.jpg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 98,
              "width": 98,
            },
          ]
        }
        testID="roundImage"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 25,
          "height": 108,
          "justifyContent": "center",
          "marginLeft": -20,
          "width": 108,
        },
      ]
    }
    testID="wrapperAvatar"
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://res.cloudinary.com/dbnq26wqe/image/upload/v1648093889/others/image_05_ww4pum.jpg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 98,
              "width": 98,
            },
          ]
        }
        testID="roundImage"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0B0B0B",
          "borderRadius": 25,
          "height": 108,
          "justifyContent": "center",
          "marginLeft": -20,
          "width": 108,
        },
      ]
    }
    testID="fillCircle"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Epilogue_500Medium",
            "fontSize": 24,
          },
        ]
      }
    >
      +
      2
    </Text>
  </View>
</View>
`;
